### ISSUE TITLE
M1 Max: DFlash2 with draft quant (w4a16) yields 0% acceptance and ~3 tok/s

### ISSUE THREAD

**Issue by imzeeshaan** [jundot/omlx#2987]
## Summary

Frustrating issue on Apple **M1 Max (64 GB)**, following the Inco DFlash 2 oMLX setup for `Qwen3.8-27B-4bit` + `Qwen3.8-27B-DFlash2` with **Draft quantization enabled** results in **0% draft acceptance** and ~**3 tok/s**.

DFlashEngine loads correctly, but speculation effectively fails: every draft token is rejected, so each cycle pays draft+verify overhead for one token. This is **slower** than plain autoregressive decode on the same machine (~11 tok/s with DFlash off).

Suspected cause(post googling): draft quant defaults to **w4a16** (`activation_bits=16`). On M1 Max this appears numerically unstable for DFlash2 (fp16 draft path), producing garbage logits and 0% acceptance. Setting `dflash_draft_quant_activation_bits: 32` (w4a32) previously restored ~60–70% acceptance and ~9 tok/s on this machine.

Any idea how to fix it..?? my LM studio gives a better ~16 tps

## Environment

- Chip: Apple M1 Max
- Unified memory: 64 GB
- macOS: 26.x (Tahoe)
- Runtime: oMLX with DFlash 2 support (manual setup per https://inco.ai/blog/dflash2/)
- Target: `mlx-community/Qwen3.8-27B-4bit`
- Draft: `incoai/Qwen3.8-27B-DFlash2`

## Config

Per Inco guide / Model Manager:

- DFlash: enabled
- Draft model: `incoai/Qwen3.8-27B-DFlash2`
- Draft quantization: enabled
- Runtime block size: `5`
- Verify mode: `dflash`

Persisted settings (relevant keys):

```json
{
  "dflash_enabled": true,
  "dflash_draft_model": ".../incoai/Qwen3.8-27B-DFlash2",
  "dflash_draft_quant_enabled": true,
  "dflash_draft_quant_weight_bits": 4,
  "dflash_draft_quant_activation_bits": 16,
  "dflash_draft_quant_group_size": 64,
  "dflash_block_size": 5,
  "dflash_verify_mode": "dflash"
}

**Comment by brechtm** [comment]
Related: #2907
